src/mono/msbuild/AppleTestRunner/AppleTestRunner.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using Microsoft.DotNet.XHarness.Tests.Runners;
+using Microsoft.DotNet.XHarness.Tests.Runners.Core;
+
+public class AppleConsole : TextWriter
+{
+    public override Encoding Encoding => Encoding.Default;
+
+    [DllImport("__Internal")]
+    public extern static void mono_ios_append_output (string value);
+
+    [DllImport("__Internal")]
+    public extern static void mono_ios_set_summary (string value);
+
+    public override void Write(string? value)
+    {
+        if (value != null)
+        {
+            mono_ios_append_output(value);
+        }
+    }
+}
+
+public class SimpleTestRunner : iOSApplicatonEntryPoint, IDevice
```

Comment:
`iOSApplicatonEntryPoint` -- is this a typo? `iOSApplicationEntryPoint`

Reply:
ah, it's in dotnet/xharness. I didn't know about this repo 😸

Reply:
cc @mandel-macaque ^

Reply:
also, while you are there could you please add events for "test is started", "test is finished (with result)" ?

Reply:
@danmosemsft I'm known for my typos, I'll get that fixed! @EgorBo  https://github.com/dotnet/xharness/issues/36